src/Compression/CompressionCodecQATZSTD.cpp

```diff
@@ -0,0 +1,114 @@
+#ifdef ENABLE_QATZSTD_COMPRESSION
+#include <Compression/CompressionInfo.h>
+#include <Compression/CompressionFactory.h>
+#include <zstd.h>
+#include <Parsers/IAST.h>
+#include <Parsers/ASTLiteral.h>
+#include <Parsers/ASTFunction.h>
+#include <Common/typeid_cast.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+#include "qatseqprod.h"
+#include "CompressionCodecZSTD.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int ILLEGAL_SYNTAX_FOR_CODEC_TYPE;
+    extern const int ILLEGAL_CODEC_PARAMETER;
+}
+
+class CompressionCodecQATZSTD : public CompressionCodecZSTD
+{
+public:
+    static constexpr auto QATZSTD_SUPPORTED_MIN_LEVEL = 1;
+    static constexpr auto QATZSTD_SUPPORTED_MAX_LEVEL = 12;
+    explicit CompressionCodecQATZSTD(int level_);
+    ~CompressionCodecQATZSTD() override;
+
+protected:
+    UInt32 doCompressData(const char * source, UInt32 source_size, char * dest) const override;
+
+private:
+    const int level;
+    mutable bool initialized;
+    mutable ZSTD_CCtx* cctx;
```

Comment:
(Cosmetic: Please go through all added code and add spaces left and right of `*` and `&`)

Reply:
Done